### PR TITLE
Clamp estimateImproperlyFollowedDifficultSliders for lazer scores

### PR DESCRIPTION
**Pull request by Finadoggie**
Fixes instances of maps giving too much pp if their slider count on the servers is less than their actual slider count (or any other scenario where a negative number of dropped slider ends is feasible)

### CODE REVIEW

**Comment by smoogipoo** [comment]
Sure, but isn't this hiding the issue?

**Comment by Finadoggie** [comment]
I guess it is. Is that a bad thing?

**Comment by Rekunan** [comment]
> isn't this hiding the issue?

It's the same for stable scores:
https://github.com/ppy/osu/blob/8bd65d9938a10fc42e6409501b0282f0fa4a25ef/osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs#L177
And existed before #27691 was merged:
https://github.com/ppy/osu/blob/970e45ff2485b8231de74b880b32ac8b97f2c098/osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs#L125

So if there is an issue with this change, then we should probably remove the clamping for stable scores (and maybe reconsider other/similar occurrences in diffcalc).

**Comment by smoogipoo** [comment]
> It's the same for stable scores:

As I read it, that's because it's an estimation where the lazer path is supposed to be more accurate. But sure.

**Review by smoogipoo — APPROVED** [review]
Approved but blocked on https://github.com/ppy/osu/pull/30566 😴

**Comment by smoogipoo** [comment]
!diffcalc

**Comment by smoogipoo** [comment]
Sorry, I'm going to use this PR as a bit of a testbed.

!diffcalc

**Comment by smoogipoo** [comment]
!diffcalc

**Comment by smoogipoo** [comment]
!diffcalc

**Pull request merged**
